2048-game.js

```diff
@@ -0,0 +1,777 @@
+const arr = [one, two, three, four, five, six, seven, eight, nine, ten, eleven, twelve, thirteen, fourteen, fifteen, sixteen];
+
+const tiles = [
+    {number: 2,
+     colour: "lavender",
+     hex: "#E6E6FA",
+     numCode: "<p>2</p>"},
+
+    {number: 4,
+     colour: "sakuraPink",
+     hex: "#FFB7C5",
+     numCode: "<p>4</p>"},
+
+    {number: 8,
+     colour: "skyBlue",
+     hex: "#87CEEB",
+     numCode: "<p>8</p>"},
+
+    {number: 16,
+     colour: "limeGreen",
+     hex: "#32CD32",
+     numCode: "<p>16</p>"},
+
+    {number: 32,
+     colour: "mustardYellow",
+     hex: "FFDB58",
+     numCode: "<p>32</p>"},
+
+    {number: 64,
+     colour: "pastelOrange",
+     hex: "#FAC898",
+     numCode: "<p>64</p>"},
+
+    {number: 128,
+     colour: "jade",
+     hex: "00A36C",
+     numCode: "<p>128</p>"},
+
+    {number: 256,
+     colour: "blushRed",
+     hex: "#BC544B",
+     numCode: "<p>256</p>"},
+
+    {number: 512,
+     colour: "pastelBlue",
+     hex: "#A7C7E7",
+     numCode: "<p>512</p>"},
+
+    {number: 1024,
+     colour: "pastelGreen",
+     hex: "#77DD77",
+     numCode: "<p>1024</p>"},
+
+    {number: 2048,
+     colour: "currantRed",
+     hex: "#680C07",
+     numCode: "<p>2048</p>"}
+]
+
+//Functions for the buttons
+function isThereAGame(){   
+    for (i=0; i < arr.length; i++){
+        if (arr[i].innerHTML.includes("2") ) {
+            return true}
+    }
+}
+
+function initiateGame(){
+    let test = 14;
+    let randomNo = Math.floor(Math.random() * 16);
+    if (isThereAGame()){
+        alert('There is already an active game!')
+    }
+    else{
+        for (i=0; i < arr.length; i++){
+            if (test == i){
+                arr[i].style.backgroundColor = `${tiles[0].hex}`;
+                arr[i].innerHTML = `<p>${tiles[0].number}</p>`;
+            }
+        }
+    }
+}
+
+function clearGame() {
+    for (i=0; i < arr.length; i++){
+        arr[i].style.backgroundColor = "";
+        arr[i].innerHTML = '';
+    }
+}
+
+
+const startButton = document.getElementById('start');
+startButton.addEventListener("click", initiateGame);
+document.addEventListener("keydown", (e) => {
+    if (e.key == 't' || e.key == 't'){
+        initiateGame();
+    }
+})
+
+const resetButton = document.getElementById('reset');
+resetButton.addEventListener("click", clearGame);
+/*document.addEventListener("keydown", (e) => {
+    if (e.key == "r" || e.key == "R"){
+        clearGame()
+    }
+});*/
+
+function openSquares(arr) {
+    let available = [];
+    for (i = 0; i < arr.length; i++){
+        if (arr[i].style.backgroundColor == ""){
+            available.push(arr[i]);
+        }
+    }
+    return available;
+}
+
+//Functions to determine if tiles can be moved in each direction
+function canItMoveRight(){
+    let counter = 0;
+    for (i = arr.length-1; i >=0; i--){
+        if (arr[i].style.backgroundColor != ""){
+            if (i == 3 || i == 7 || i == 11 || i == 15){
```

Comment:
You can do 
`if ([3, 7, 11, 15].includes(i)) { `

in python it would be `if i in [3,7,11,15]: ` which makes more human sense

Then you have a variable called edges the next line, you can use that variable for this so it becomes
`if (edges.include(i))` which is a lot cleaner and self documents what it's doing